### ISSUE TITLE
timeRangeError is not triggered

### ISSUE THREAD

**Issue by zahirdhada** [jonthornton/jquery-timepicker#531]
As specified in the title, 'timeRangeError' event is not triggered. I looked at the code and found following codition:

``` javascript
// check that the time in within bounds
if (settings.minTime !== null && seconds < settings.minTime 
    && settings.maxTime !== null && seconds > settings.maxTime) {
        rangeError = true;
}
```

The condition will never evaluate to true because `seconds` cannot be both less than `minTime` and greater than `maxTime` at the same time when `minTime < maxTime`.

I changed the above block of code as following:

``` javascript
// check that the time in within bounds
var minTimeNormal = null;
var maxTimeNormal = null;
var secondsNormal = seconds;

if( settings.minTime !== null) {
    minTimeNormal = settings.minTime;
}
if( settings.maxTime !== null) {
    maxTimeNormal = settings.maxTime;
    if(minTimeNormal !== null && minTimeNormal > maxTimeNormal) {
        maxTimeNormal += _ONE_DAY;
        if(secondsNormal < minTimeNormal)
            secondsNormal += _ONE_DAY;
    }
}

if ((minTimeNormal !== null && maxTimeNormal !== null) 
        && (secondsNormal < minTimeNormal || secondsNormal > maxTimeNormal)) {
    rangeError = true;
}
```

It also takes care of the case where `minTime > maxTime`.

Notice: Although it is working for me, the code is not fully tested.

**Comment by zahir-rg** [comment]
I believe this is now closed by #552

**Comment by zahirdhada** [comment]
Closed by #552

**Comment by zahirdhada** [comment]
Closed by #552
